packages/components/src/ui/data-table-filter/core/types.ts

```
@@ -0,0 +1,305 @@
+export type TTransformOptionFn<TVal = unknown> = (value: ElementType<NonNullable<TVal>>) => ColumnOption;
+
+/*
+ * Used by `option` and `multiOption` columns.
+ * A custom ordering function when sorting a column's options.
+ */
+export type TOrderFn<TVal = unknown> = (a: ElementType<NonNullable<TVal>>, b: ElementType<NonNullable<TVal>>) => number;
+
+/*
+ * The configuration for a column.
+ */
+export type ColumnConfig<TData, TType extends ColumnDataType = any, TVal = unknown, TId extends string = string> = {
```

Biome (1.9.4)

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)